x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/WaterfallItem.tsx

```diff
@@ -7,39 +7,81 @@
 import React from 'react';
 import styled from 'styled-components';
 
+import { EuiIcon } from '@elastic/eui';
 import {
   colors,
+  fontFamily,
   fontFamilyCode,
+  fontSize,
   fontSizes,
   px,
   unit,
   units
 } from '../../../../../../style/variables';
 import { IWaterfallItem } from './waterfall_helpers/waterfall_helpers';
 
-const ItemBar = styled.div`
+interface ItemBarProps {
+  type: 'transaction' | 'span';
+  left: number;
+  width: number;
+  color: string;
+}
+
+interface ITimelineMargins {
+  right: number;
+  left: number;
+  top: number;
+  bottom: number;
+}
+
+interface IWaterfallItemProps {
+  timelineMargins: ITimelineMargins;
+  totalDuration: number;
+  item: IWaterfallItem;
+  color: string;
+  isSelected: boolean;
+  onClick: () => any;
+}
+
+type IContainerProps = Pick<
+  IWaterfallItemProps,
+  'item' | 'timelineMargins' | 'isSelected'
+>;
+
+const ItemBar = styled<ItemBarProps, any>('div')`
+  box-sizing: border-box;
   position: relative;
-  height: ${unit}px;
+  height: px(${unit});
```

Comment:
Let's try this suggestion thing again :D 
```suggestion
  height: ${px(unit)};
```